src/services/rest/http-client.js

```diff
@@ -132,13 +135,19 @@ export default class HttpClient {
     const settings = createRequestSettings(restOpts, opts);
     const req = this.httpModule.request(settings);
 
-    if (typeof opts.body !== 'undefined') {
+    const hasBody = typeof opts.body !== 'undefined';
+    if (hasBody) {
       req.write(opts.body);
     }
 
+    if (restOpts.handleLog) {
+      restOpts.handleLog(hasBody ?
```

Comment:
Maybe `{ msg: 'Sent', data: { request: settings, body: opts.body || null } }`? Or is it important to not expose a `body` property if it doesn't exist?